src/Microsoft.AspNetCore.Razor.Evolution/CodeGeneration/DesignTimeCSharpRenderer.cs

```diff
@@ -260,13 +260,10 @@ public override void VisitSetTagHelperProperty(SetTagHelperPropertyIRNode node)
                 VisitDefault(node);
 
                 Context.Writer.WriteStartAssignment(propertyValueAccessor);
-                if (node.Children.Count == 1 && node.Children.First() is HtmlContentIRNode)
+                if (node.Children.Count == 1 && node.Children.First() is HtmlContentIRNode htmlNode)
                 {
-                    var htmlNode = node.Children.First() as HtmlContentIRNode;
-                    if (htmlNode != null)
-                    {
-                        Context.Writer.WriteStringLiteral(htmlNode.Content);
-                    }
+                    var content = GetContent(htmlNode);
```

Comment:
This and a few others should go away when https://github.com/aspnet/Razor/issues/963 is fixed.